src/api/define.js

```diff
@@ -102,10 +102,19 @@ function generateUniqueName(name) {
 }
 
 function registerUniqueName(name) {
-  registry[name] = registry[name] ? registry[name] + 1 : 1;
+  if (registry[name]) {
+    registerUniqueName(name + '-'+ registry[name]);
```

Comment:
The only concern with uuid (and other implementations) is size and performance. That library is almost 300 LoC which is significant weight when Skate itself is only 1200 LoC. I'd be interested to see if something like http://stackoverflow.com/a/2117523/110840 could work for this. My only concern there is degrading registration performance.